cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -75,7 +78,17 @@ func (c *CStorVolumeReplicaController) syncHandler(
 			key,
 		)
 	}
-
+	cvrObj, err := c.populateVersion(cvrGot)
+	if err != nil {
+		klog.Errorf("failed to add versionDetails to cstorvolumereplica %s:%s", cvrGot.Name, err.Error())
+		return err
+	}
+	cvrObj, err = c.upgrade(cvrObj)
```

Comment:
instead of upgrade, how about calling it as `reconcileVersion`?

Reply:
the task of upgrade function is to perform steps to reconcile the version. Let us keep it as upgrade for now.